yazi-core/src/input/input.rs

```diff
@@ -55,12 +102,79 @@ impl Input {
 			return self.type_str(c.encode_utf8(&mut bits));
 		}
 
+		use KeyCode::{Backspace, Char as C, Delete};
+
 		match key {
-			Key { code: KeyCode::Backspace, shift: false, ctrl: false, alt: false } => self.backspace(),
+			Key { code: Backspace, shift: false, ctrl: false, alt: false } => self.backspace(),
+			// Handle Emacs-style keybindings.
+			Key { code: C('a'), shift: false, ctrl: true, alt: false } => self.move_(isize::MIN),
+			Key { code: C('e'), shift: false, ctrl: true, alt: false } => self.move_(isize::MAX),
+			Key { code: C('b'), shift: false, ctrl: true, alt: false } => self.move_(-1),
+			Key { code: C('f'), shift: false, ctrl: true, alt: false } => self.move_(1),
+			Key { code: C('h'), shift: false, ctrl: true, alt: false } => self.backspace(),
+			Key { code: C('d'), shift: false, ctrl: true, alt: false } => self.forward_delete(),
+			Key { code: C('b'), shift: false, ctrl: false, alt: true } => {
+				let snap = self.snap();
+				let idx = snap.idx(snap.cursor).unwrap_or(snap.len());
+
+				let step = Self::find_word_boundary(snap.value[..idx].chars().rev());
+				self.move_(-(step as isize))
+			}
+			Key { code: C('f'), shift: false, ctrl: false, alt: true } => {
+				let snap = self.snap();
+				let idx = snap.idx(snap.cursor).unwrap_or(snap.len());
+
+				let step = Self::find_word_boundary(snap.value[idx..].chars());
+				self.move_(step as isize)
+			}
+			Key { code: C('u'), shift: false, ctrl: true, alt: false } => {
+				let snap = self.snap_mut();
+				let end = snap.idx(snap.cursor).unwrap_or(snap.len());
+				self.delete_range(..end);
+				true
+			}
+			Key { code: C('k'), shift: false, ctrl: true, alt: false }
+			| Key { code: Delete, shift: false, ctrl: false, alt: false } => {
```

Comment:
Is this `Delete` should be added for `Ctrl-u` instead of `Ctrl-k`? If so I will fix it.

Reply:
I `believe` that Delete cuts to the end of line, based off what iTerm sends. I'll look into it real quick

Reply:
This is from nushell's listening to keybinds.

<img width="518" alt="image" src="https://github.com/sxyazi/yazi/assets/70821802/2efe8ff9-5846-4e1a-8906-15efae956729">

In `reedline`, however, the `Delete` key maps to forward delete.

I'm not sure now what the behavior should be. Perhaps it would be better to leave `Delete` out.

Reply:
Okay I'll remove it first